app/Http/Controllers/UserManagementController.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Interns2025b\Http\Controllers;
+
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Hash;
+use Interns2025b\Actions\RegisterUserAction;
+use Interns2025b\Enums\Role;
+use Interns2025b\Http\Requests\StoreUserRequest;
+use Interns2025b\Http\Requests\UpdateUserRequest;
+use Interns2025b\Http\Resources\UserResource;
+use Interns2025b\Models\User;
+use Symfony\Component\HttpFoundation\Response as Status;
+
+class UserManagementController extends Controller
+{
+    public function index(): JsonResponse
+    {
+        $this->authorize("viewAny", User::class);
+
+        $users = User::query()
+            ->with("organizations")
+            ->role(Role::User->value)
+            ->orderBy("id")
+            ->get();
+
+        return response()->json(UserResource::collection($users), Status::HTTP_OK);
+    }
+
+    public function show(User $user): JsonResponse
+    {
+        $this->authorize("view", $user);
+
+        $user->load("organizations");
+
+        return response()->json(new UserResource($user), Status::HTTP_OK);
+    }
+
+    public function store(StoreUserRequest $request, RegisterUserAction $registerUser): JsonResponse
+    {
+        $this->authorize("create", User::class);
+
+        $dto = $request->toDto();
+
+        $user = $registerUser->execute($dto);
+
+        if (!$user) {
+            return response()->json([
+                "message" => __("users.email_exists"),
+            ], Status::HTTP_CONFLICT);
+        }
+
+        $user->syncRoles(Role::User->value);
+
+        if ($request->filled("organization_ids")) {
+            $user->organizations()->sync($request->input("organization_ids"));
+        }
+
+        return response()->json(new UserResource($user), Status::HTTP_CREATED);
+    }
+
+    public function update(UpdateUserRequest $request, User $user): JsonResponse
+    {
+        $this->authorize("update", $user);
+
+        $dto = $request->toDto();
+
+        $emailChanged = $dto->email !== null && $dto->email !== $user->email;
+
+        $updateData = array_filter([
+            "first_name" => $dto->firstName,
+            "last_name" => $dto->lastName,
```

Comment:
Can you set `last_name` as **null** now with this filtering?  All existing data should be sent when updating, so this filtering seems unnecessary